### ISSUE TITLE
Feature flag request: Separate argument injection from snippets

### ISSUE THREAD

**Issue by peteruhnak** [zigtools/zls#685]
Would it be possible to separate argument injection into its own config independent of `enable_snippets`?

With `enable_snippets` enabled, accepting a function completion also injects the function's args and types,

e.g.

```zig
exe.addCS
```
is completed to

```zig
exe.addCSourceFile(file: []const u8, flags: []const []const u8)
```

instead of

```zig
exe.addCSourceFile()
```

I'm sure this is quite useful to some people and some code editors, but it seems like a separate functionality to be toggled independently (and to me it is quite annoying).

So basically change the code here to use different flag, and add docs for it & everything https://github.com/zigtools/zls/blob/8eeae0286559a5765adb55d82c4b2046786fcc58/src/Server.zig#L498

I can create a PR if you are ok with the change.

**Comment by Techatrix** [comment]
Sounds like a good idea.
Because its sometimes easy to forget editing all locations for a new config options, i will list them here which may help you:
1. https://github.com/zigtools/zls/blob/8cf93fc96a705d30771843de6415ddfce097022f/src/Config.zig#L14
2. https://github.com/zigtools/zls/blob/8cf93fc96a705d30771843de6415ddfce097022f/src/requests.zig#L290
3. https://github.com/zigtools/zls/blob/e776a5c8e3945d19132e86754cc241789e3b0133/schema.json#L7
4. https://github.com/zigtools/zls/blob/a70beba6d26387b5e24963c0348b13e72dacc259/README.md#L106
5. https://github.com/zigtools/zls/blob/e776a5c8e3945d19132e86754cc241789e3b0133/src/setup.zig#L172
